types/jasmine/jasmine-tests.ts

```diff
@@ -928,37 +928,101 @@ describe("A spy, when created manually", () => {
 });
 
 describe("a spy on a typed method", () => {
-    class Test {
-        method(arg: number): string {
-            return "42";
+    describe("method parameter is primitive", () => {
+        class Test {
+            method(arg: number): string {
+                return "42";
+            }
         }
-    }
 
-    let t: Test;
+        let t: Test;
 
-    beforeEach(() => {
-        t = new Test();
+        beforeEach(() => {
+            t = new Test();
+        });
+
+        it("should match only call arguments with the correct type", () => {
+            const spy = spyOn(t, "method");
+            t.method(1);
+
+            expect(t.method).toHaveBeenCalledWith(1);
+            expect(t.method).toHaveBeenCalledWith("1"); // $ExpectError
+            expect(t.method).not.toHaveBeenCalledWith("1"); // $ExpectError
+
+            expect(spy).toHaveBeenCalledWith(1);
+            expect(spy).toHaveBeenCalledWith("1"); // $ExpectError
+            expect(spy).not.toHaveBeenCalledWith("1"); // $ExpectError
+
+            expect(t.method).toHaveBeenCalledOnceWith(1);
+            expect(t.method).toHaveBeenCalledOnceWith("1"); // $ExpectError
+            expect(t.method).not.toHaveBeenCalledOnceWith("1"); // $ExpectError
+
+            expect(spy).toHaveBeenCalledOnceWith(1);
+            expect(spy).toHaveBeenCalledOnceWith("1"); // $ExpectError
+            expect(spy).not.toHaveBeenCalledOnceWith("1"); // $ExpectError
+        });
     });
 
-    it("should match only call arguments with the correct type", () => {
-        const spy = spyOn(t, "method");
-        t.method(1);
+    describe("method parameter is an object", () => {
+        interface ObjectProperty {
+            prop1: string;
+            prop2: number;
+        }
 
-        expect(t.method).toHaveBeenCalledWith(1);
-        expect(t.method).toHaveBeenCalledWith("1"); // $ExpectError
-        expect(t.method).not.toHaveBeenCalledWith("1"); // $ExpectError
+        interface ObjectParam {
+            prop1: string;
+            prop2: ObjectProperty;
+        }
 
-        expect(spy).toHaveBeenCalledWith(1);
-        expect(spy).toHaveBeenCalledWith("1"); // $ExpectError
-        expect(spy).not.toHaveBeenCalledWith("1"); // $ExpectError
+        class Test {
+            method(arg: ObjectParam): string {
+                return "42";
+            }
+        }
 
-        expect(t.method).toHaveBeenCalledOnceWith(1);
-        expect(t.method).toHaveBeenCalledOnceWith("1"); // $ExpectError
-        expect(t.method).not.toHaveBeenCalledOnceWith("1"); // $ExpectError
+        let t: Test;
 
-        expect(spy).toHaveBeenCalledOnceWith(1);
-        expect(spy).toHaveBeenCalledOnceWith("1"); // $ExpectError
-        expect(spy).not.toHaveBeenCalledOnceWith("1"); // $ExpectError
+        beforeEach(() => {
+            t = new Test();
+        });
+
+        it("should match only call arguments with the correct type", () => {
```

Comment:
Could you show me what the results looked like for this test before your PR?

Reply:
Reverting my change and running the tests has this output:
```
ERROR: 1014:53   expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'string'.
ERROR: 1014:111  expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'number'.
ERROR: 1017:48   expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'string'.
ERROR: 1017:106  expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'number'.
ERROR: 1020:57   expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'string'.
ERROR: 1020:115  expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'number'.
ERROR: 1023:52   expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'string'.
ERROR: 1023:110  expect  TypeScript@4.4 compile error: 
Type 'AsymmetricMatcher<any>' is not assignable to type 'number'.
```

Reply:
All right, thanks. I don't fully understand what's going on here, but considering you have pretty thorough tests and the tests look correct, I can't argue with it.